crates/cxx-qt-gen/src/syntax/qtfile.rs

```diff
@@ -41,7 +41,12 @@ impl ToTokens for CxxQtFile {
 }
 
 pub fn parse_qt_file(path: impl AsRef<std::path::Path>) -> Result<CxxQtFile> {
-    let source = std::fs::read_to_string(path.as_ref()).expect("Could not read path {} to string");
+    let source = match std::fs::read_to_string(path.as_ref()) {
+        Ok(source) => source,
+        Err(err) => {
+            panic!("Failed to read file {:?}: {}", path.as_ref(), err);
+        }
+    };
```

Comment:
I think `match` is more verbose than needed here. I think we can keep the original `.expect` and use `&format!("Could not read path {} to string", path.as_ref().display())`

Reply:
I do find it interesting that Rust doesn't have a function pointer route for expect, like how unwrap has `unwrap_or_else` vs `unwrap_or` so you can have lazy evaluation. Eg it would be nice if there was `expect_slow(|| my_slow_except_thing())`.

Unless, can we hack things with `unwrap_or_else(|| panic!(&format!(...)))` or is that crazy? :laughing: 

As for returning an error, we are currently using `syn::Result`, so it would need a span, and this error occurs before we have any tokens to span with so could be tricky :sweat_smile: Maybe later we'll have our own Error with a enum of IOError / Syn etc but we haven't got that far yet, so this can just panic for now.

Reply:
`unwrap_or_else` can have weird behavior in the callstack, anyway the best choice here would be to implement an enum Error type, as you said, but you can still for now use [anyhow - Rust](https://docs.rs/anyhow/1.0.70/anyhow/) crate's `Result` type.

Implementing a custom error type seems like a good issue.

Reply:
I'm not concerned about performance here so I think the simplest solution would be best: `.expect(&format!("Could not read path {} to string", path.as_ref().display()))`

Reply:
> Unless, can we hack things with unwrap_or_else(|| panic!(&format!(...))) or is that crazy? laughing

Searching for any discussion of an `expect_or_else` function in https://github.com/rust-lang/rust, I found https://github.com/rust-lang/rust/commit/fe588d894fa392fdf787c18a959b342d38d0c71c. So I think that would be an okay solution.

Reply:
> `unwrap_or_else` can have weird behavior in the callstack, anyway the best choice here would be to implement an enum Error type, as you said, but you can still for now use [anyhow - Rust](https://docs.rs/anyhow/1.0.70/anyhow/) crate's `Result` type.
> 
> Implementing a custom error type seems like a good issue.

Right, this would be more a medium/long term goal as a separate issue :+1:  so far `syn::Result` has been good enough for most errors as we want to refer to a span most of the time anyway.

For this pull request, either of the solutions (expect, match, unwrap_or_else) are likely fine as discussed, things can be refactored again later when/if we have our own error type.

Reply:
Of course if someone is able to get a custom error type working nicely, then feel free to open another pull request :-)

Reply:
So I forced pushed unwrap_or_else, and added a todo comment so we know it's not the best fix anyway, but at least it will give the user all the hint to debug why his build system doesn't work, which is the most important for now